src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -98,6 +109,33 @@ internal static class ParameterHelpers
             return parameters;
         }
 
+        private static bool IsPropertyWithoutBody(Symbol owner)
+        {
+            if (owner is SourcePropertySymbol)
```

Comment:
> Does it have to be a SourcePropertySymbol or can we cover all PropertySymbols?

How would you have readonly keyword on an indexer that wasn't a source indexer?

> usually, do Kind check instead of type check whenever possible.  

How do you do a kind check that this is a SourcePropertySymbol?

> avoid doing type check twice by storing result of an as Type operation into a local and then checking it for null

Sure, i can change that.